### ISSUE TITLE
BigQuery Storage API integrations should accept either v1beta1 or v1 client

### ISSUE THREAD

**Issue by tswast** [googleapis/python-bigquery#84]
I notice that the v1beta1 API is still used. https://github.com/googleapis/python-bigquery/blob/952164b0f7073ffd06b6f7057dafcf6526ebb95a/google/cloud/bigquery/_pandas_helpers.py#L577-L587

This is good, as it keeps packages such as pandas-gbq from breaking, but shouldn't we start thinking about migrating folks to the v1 endpoint? To do that without causing too much disruption, the `to_dataframe`, magics, and DB-API integrations should accept either v1beta1 or v1 clients for an extended period of time.

**Comment by plamut** [comment]
@tswast Now that the BQ Storage client is stable, #55 will start using it as a default, and also migrate to v1 stable from v1beta1.

If this would represent to big a disruption, the PR needs to be adjusted to support both Storage client versions. Possibly with a warning if beta client is used?

Also feel free to comment on the PR, if you have time for it.

**Comment by tswast** [comment]
Yes, that would be a very big disruption. Definitely a breaking change, as it would break pandas-gbq among others.

**Comment by plamut** [comment]
Noted. We will add this then, and branch on the BQ Storage client type (and if v1beta1 is passed, the old logic for constructing the session will be used).
